#VRML_SIM R2020a utf8
WorldInfo {
  gpsCoordinateSystem "WGS84"
}
Viewpoint {
  orientation 0.25417480859135966 0.9492940863837198 0.18502946855688832 4.942553511041744
  position -3.129103149412509 1.5084307463116469 0.9835219271698284
  follow "SUGV:body"
  followType "Mounted Shot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
  floorSize 20 20
}
Solid {
  translation 0 0.12 1.05
  children [
    Shape {
      appearance Appearance {
        material Material {
          diffuseColor 0 0.1 0.3
        }
      }
      geometry Box {
        size 0.1 0.1 0.1
      }
    }
  ]
  boundingObject Box {
    size 0.1 0.1 0.1
  }
  physics Physics {
  }
  recognitionColors [
    0 0.1 0.3
  ]
}
DEF SUGV Robot {
  translation 0 0.16 0
  children [
    Camera {
      translation 0 0.0499992 0.16
      rotation 0 1 0 3.14159
      children [
        DEF camera Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Box {
            size 0.015 0.015 0.015
          }
        }
      ]
      boundingObject USE camera
      physics Physics {
      }
      fieldOfView 1.0472
      width 1296
      height 976
      recognition Recognition {
      }
    }
    TouchSensor {
      translation 0 0 0.16
      scale 0.458998 0.458998 0.458998
      children [
        DEF Bumper Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0 0
            }
          }
          geometry Box {
            size 0.1 0.1 0.05
          }
        }
      ]
      boundingObject USE Bumper
      physics Physics {
      }
      translationStep 0.001
    }
    Compass {
      yAxis FALSE
    }
    GPS {
    }
    DEF Wheel_LB HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.13 -0.0499998 -0.11
      }
      device [
        RotationalMotor {
          name "Motor_LB"
        }
      ]
      endPoint Solid {
        translation 0.13 -0.0499998 -0.11
        rotation 0 0 1 1.5708
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 1
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_LB"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_RM HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.13 -0.05 0
      }
      endPoint Solid {
        translation 0.13 -0.05 0
        rotation 0 0 1 1.5708
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_RM"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_LF HingeJoint {
      jointParameters HingeJointParameters {
        anchor 0.13 -0.0500003 0.11
      }
      device [
        RotationalMotor {
          name "Motor_LF"
        }
      ]
      endPoint Solid {
        translation 0.13 -0.0500003 0.11
        rotation 0 0 1 1.5708
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_LF"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_RB HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.13 -0.0500008 -0.11
      }
      device [
        RotationalMotor {
          name "Motor_RB"
        }
      ]
      endPoint Solid {
        translation -0.13 -0.0500008 -0.11
        rotation 0 0 1 1.5708
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 1
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_RB"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_LM HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.13 -0.0500008 0
      }
      endPoint Solid {
        translation -0.13 -0.0500008 0
        rotation 0 0 1 1.5708
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_LM"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Wheel_RF HingeJoint {
      jointParameters HingeJointParameters {
        anchor -0.13 -0.0500008 0.11
      }
      device [
        RotationalMotor {
          name "Motor_RF"
        }
      ]
      endPoint Solid {
        translation -0.13 -0.0500008 0.11
        rotation 0 0 1 1.5708
        children [
          Shape {
            appearance Appearance {
              material Material {
                diffuseColor 0 1 0
              }
            }
            geometry Cylinder {
              height 0.0254
              radius 0.0381
            }
          }
        ]
        name "Wheel_RF"
        boundingObject Cylinder {
          height 0.0254
          radius 0.0381
        }
        physics Physics {
        }
      }
    }
    DEF Body Solid {
      translation 0 0.0178 0
      children [
        Shape {
          appearance Appearance {
            material Material {
              diffuseColor 1 0 0
            }
          }
          geometry Box {
            size 0.2286 0.16 0.3048
          }
        }
      ]
      name "body"
      boundingObject Box {
        size 0.2286 0.16 0.3048
      }
      physics Physics {
      }
    }
  ]
  name "SUGV"
  boundingObject Transform {
    translation 0 0.0178 0
    children [
      Box {
        size 0.2286 0.16 0.3048
      }
    ]
  }
  physics Physics {
  }
  controller "<extern>"
}
